microsoft/testsuites/kselftest/kselftest.py

```diff
@@ -0,0 +1,263 @@
+
+        # tar file path specified in yml
+        self._tar_file_path = kselftest_file_path
+        if self._tar_file_path:
+            self._remote_tar_path = self.get_tool_path(
+                use_global=True
+            ) / os.path.basename(self._tar_file_path)
+        else:
+            self._remote_tar_path = (
+                self.get_tool_path(use_global=True)
+                / f"{self._KERNEL_REPO_NAME}{self._KSELFTEST_TAR_PATH}"
+            )
+
+        # command to run kselftests
+        self._command = (
+            self.get_tool_path(use_global=True) / "kselftest-packages/run_kselftest.sh"
+        )
+
+    # install common dependencies
+    def _install(self) -> bool:
+        if not isinstance(self.node.os, CBLMariner):
+            raise SkippedException("kselftests are supported on Mariner VMs only.")
+
+        if self._tar_file_path:
+            self.node.shell.copy(PurePath(self._tar_file_path), self._remote_tar_path)
+        else:
+            # clone kernel, build kernel, then build kselftests
+            self.node.os.install_packages(
+                [
+                    "bison",
+                    "flex",
+                    "build-essential",
+                    "openssl-devel",
+                    "bc",
+                    "dwarves",
+                    "rsync",
+                    "libcap-devel",
+                    "libcap-ng-devel",
+                    "fuse",
+                    "fuse-devel",
+                    "popt-devel",
+                    "numactl-devel",
+                    "libmnl-devel",
+                    "libinput",
+                    "mesa-libgbm-devel",
+                    "glibc-static",
+                    "clang",
+                ]
+            )
+
+            uname = self.node.tools[Uname]
+            uname_result = uname.get_linux_information(force_run=False)
+            version = uname_result.kernel_version
+            git = self.node.tools[Git]
+
+            # If version.patch is zero, clone major.minor kernel from
+            # upstream stable kernel. If version.patch is non-zero, clone
+            # major.minor.patch kernel corresponding to the distro in use
+            branch = "rolling-lts/mariner-2/"
+            branch += f"{version.major}.{version.minor}.{version.patch}.1"
+            branch_to_clone = f"{self._MARINER_KERNEL_SRC_REPO} -b {branch} --depth 1"
+            kernel_path = git.clone(
+                branch_to_clone,
+                self.get_tool_path(use_global=True),
+                fail_on_exists=False,
+            )
+            self.node.tools[Cp].copy(
+                src=self.node.get_pure_path(
+                    f"/boot/config-{uname_result.kernel_version_raw}"
+                ),
+                dest=PurePath(".config"),
+                cwd=kernel_path,
+                sudo=True,
+            )
+
+            # build kselftests
+            self.node.tools[Make].run(
```

Comment:
please check whether it compile successfully or not

Reply:
> please check whether it compile successfully or not

Added expected exit code, failure message and assert_exit_code().